nifi-system-tests/nifi-system-test-suite/src/test/java/org/apache/nifi/tests/system/NiFiClientUtil.java

```diff
@@ -395,6 +399,153 @@ public void deleteReportingTasks() throws NiFiClientException, IOException {
+        runStatusEntity.setState("ENABLED");
+        runStatusEntity.setRevision(entity.getRevision());
+        runStatusEntity.setDisconnectedNodeAcknowledged(true);
+
+        return nifiClient.getControllerClient().activateFlowAnalysisRule(entity.getId(), runStatusEntity);
+    }
+
+    public FlowAnalysisRuleEntity disableFlowAnalysisRule(final FlowAnalysisRuleEntity entity) throws NiFiClientException, IOException {
+        final FlowAnalysisRuleRunStatusEntity runStatusEntity = new FlowAnalysisRuleRunStatusEntity();
+        runStatusEntity.setState("DISABLED");
+        runStatusEntity.setRevision(entity.getRevision());
+        runStatusEntity.setDisconnectedNodeAcknowledged(true);
+
+        return nifiClient.getControllerClient().activateFlowAnalysisRule(entity.getId(), runStatusEntity);
+    }
+
+    public void disableFlowAnalysisRues() throws NiFiClientException, IOException {
+        final FlowAnalysisRulesEntity rules = nifiClient.getControllerClient().getFlowAnalysisRules();
+
+        Collection<String> toBeDisabledRuleIds = new ArrayList<>();
+        for (final FlowAnalysisRuleEntity rule : rules.getFlowAnalysisRules()) {
+            disableFlowAnalysisRule(rule);
+            toBeDisabledRuleIds.add(rule.getId());
+        }
+
+        waitForFlowAnalysisRuleState("DISABLED", toBeDisabledRuleIds);
+    }
+
+    public void deleteFlowAnalysisRules() throws NiFiClientException, IOException {
+        final FlowAnalysisRulesEntity tasksEntity = nifiClient.getControllerClient().getFlowAnalysisRules();
+        for (final FlowAnalysisRuleEntity taskEntity : tasksEntity.getFlowAnalysisRules()) {
+            taskEntity.setDisconnectedNodeAcknowledged(true);
+            nifiClient.getControllerClient().deleteFlowAnalysisRule(taskEntity);
+        }
+    }
+
+    public void waitForFlowAnalysisRulesDisabled(final String... ruleIdsOfInterest) throws NiFiClientException, IOException {
+        waitForFlowAnalysisRuleState("DISABLED", Arrays.asList(ruleIdsOfInterest));
+    }
+
+    public void waitForFlowAnalysisRulesEnabled(final String... ruleIdsOfInterest) throws NiFiClientException, IOException {
+        waitForFlowAnalysisRuleState("ENABLED", Arrays.asList(ruleIdsOfInterest));
+    }
+
+    public void waitForFlowAnalysisRulesEnabled(final List<String> ruleIdsOfInterest) throws NiFiClientException, IOException {
+        waitForFlowAnalysisRuleState("ENABLED", ruleIdsOfInterest);
+    }
+
+    public void waitForFlowAnalysisRuleState(final String desiredState, final Collection<String> ruleIdsOfInterest) throws NiFiClientException, IOException {
+        final long maxTimestamp = System.currentTimeMillis() + TimeUnit.MINUTES.toMillis(2L);
+
+        while (System.currentTimeMillis() < maxTimestamp) {
+            final List<FlowAnalysisRuleEntity> nonDisabledRules = getFlowAnalysisRulesNotInState(desiredState, ruleIdsOfInterest);
```

Comment:
` nonDisabledRules` has a wrong name because `desiredState` can be `ENABLED` or `DISABLED`.
Maybe `nonDesiredStateRules`?